admin/starter/tests/_support/DatabaseTestCase.php

```diff
@@ -5,10 +5,7 @@
 use CodeIgniter\Test\CIUnitTestCase;
 use CodeIgniter\Test\DatabaseTestTrait;
 
-/**
- * @internal
- */
-final class DatabaseTestCase extends CIUnitTestCase
+abstract class DatabaseTestCase extends CIUnitTestCase
```

Comment:
The `DatabaseTestCase` class is deprecated. Add `@deprecated`, please.

And also update `ExampleDatabaseTest` not to use it.

Reply:
You mean `CIDatabaseTestCase`? https://github.com/codeigniter4/CodeIgniter4/blob/develop/system/Test/CIDatabaseTestCase.php

I cannot recall this is also deprecated.

Reply:
Hmm. I think no. This is the relevant PR: https://github.com/codeigniter4/CodeIgniter4/pull/4374

The upgrading guide seems to be wrong here.

Reply:
Maybe. But sample code for database testing also does not use `DatabaseTestCase`.
https://codeigniter4.github.io/CodeIgniter4/testing/database.html#the-test-class

And the following code should be removed, because it is in `CIUnitTestCase`.
https://github.com/codeigniter4/CodeIgniter4/blob/a0c30ced36bd06dc7b63637ea1b549a52692fc40/admin/starter/tests/_support/DatabaseTestCase.php#L11-L57

Reply:
The deprecation should have read `CIDatabaseTestCase`. That said these classes are old and definitely merit updating. Because these are templates to be copied in and are not directly namespaced we can change anything we want. I would be favor in the big picture of moving most of **admin/** out into separate repos, but for now let's just apply the abstract and clean up the duplicate properties.

Reply:
@MGatner 
> The deprecation should have read CIDatabaseTestCase.

Okay. I sent PR to fix it. #5169